flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -331,6 +331,22 @@ public void testListAFileFailsBecauseDirectoryIsExpected() throws IOException {
         FileUtils.listFilesInDirectory(file.toPath(), FileUtils::isJarFile);
     }
 
+    @Test
+    public void testFollowSymbolicLink() throws IOException {
+        final String fileName = "a/a.jar";
+        File folder = tmp.newFolder("a");
+        final File file = tmp.newFile(fileName);
+
+        java.nio.file.Path linkPath = Paths.get(folder.getParent(), "a.lnk");
+        Files.createSymbolicLink(linkPath, file.getParentFile().toPath());
+        Collection<java.nio.file.Path> paths =
+                FileUtils.listFilesInDirectory(linkPath.toRealPath(), FileUtils::isJarFile);
```

Comment:
Doesn't `toRealPath` resolves the symbolic link already? Then this test is not really testing the new functionality contained in `listFilesInDirectory`.

Reply:
`listFilesInDirectory` resolves symbolic links for files under given path which name is still symbolic.
`toRealPath` resolve the top symbolic path only.
I have already changed test code to prove it.